### ISSUE TITLE
Broad semantic categories (animals, body parts) don't cluster well

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#149]
## Summary

Root embeddings don't capture broad taxonomic categories like "animals" or "body parts" well enough. Words within these categories have low mutual similarity.

## Test Results

```
TestCategoryClustering::test_fundamento_animals FAILED
  Animals should cluster: mean_sim=0.145 (expected > 0.3)

TestCategoryClustering::test_fundamento_body_parts FAILED
  Body parts should cluster: mean_sim=0.198 (expected > 0.3)
```

## What Works

- Colors clustering: **PASS** (mean_sim > 0.3)
- Kinship clustering: **PASS** (mean_sim > 0.3)
- Synonyms (ReVo): **PASS**
- Antonyms (ReVo): **PASS**

## Analysis

The training data (ReVo definitions + Ekzercaro co-occurrence) provides:
- Strong synonym/antonym signal from ReVo
- Good co-occurrence for common word pairs

But lacks:
- Explicit taxonomic hierarchy (hund IS-A besto)
- Category membership signal

## Root Cause

Current training uses:
1. **Definition similarity**: Words with similar definitions are similar
2. **Co-occurrence**: Words appearing together are similar

Neither directly encodes "hund and kat are both animals".

### Option A: Add ReVo hypernym loss (recommended)
Use existing ReVo hypernym/hyponym relations:
```python
# We have 3,351 hypernym + 1,098 hyponym pairs
for (hyponym, hypernym) in revo_hypernyms:
    loss += contrastive_loss(embed(hyponym), embed(hypernym), positive=True)
```

### Option B: Add category co-training
Create category labels from WordNet/Fundamento categories:
```python
categories = load_semantic_categories()
for category in categories:
    members = category.members  # e.g., [hund, kat, bird, ...]
    loss += cluster_loss(embed(members))
```

### Option C: Lower test thresholds
Accept that 0.15-0.20 similarity is "clustered enough" for practical use.

## Data Available

From `data/semantic_categories.json`:
- 22 Fundamento categories with ~300 roots
- 10 WordNet categories with ~150 roots
- ReVo has 3,351 hypernym pairs

## Priority

Medium - affects category-based retrieval but current model works for other use cases.

## Labels
- `training`
- `enhancement`
